pkg/frontend/querymiddleware/split_by_instant_interval.go

```diff
@@ -134,10 +137,18 @@ func (s *splitInstantQueryByIntervalMiddleware) Do(ctx context.Context, req Requ
 		return s.next.Do(ctx, req)
 	}
 
-	if stats.GetSplitQueries() == 0 {
+	if stats.GetSplitQueries() <= 0 {
 		// the query cannot be split, so continue
 		level.Debug(spanLog).Log("msg", "input query resulted in a no operation, falling back to try executing without splitting")
-		s.metrics.splittingSkipped.WithLabelValues(skippedReasonNoop).Inc()
+		if stats.GetNoOpSmallIntervalQuery() {
+			s.metrics.splittingSkipped.WithLabelValues(skippedReasonNoopSmallInterval).Inc()
+		}
+		if stats.GetNoOpSubquery() {
+			s.metrics.splittingSkipped.WithLabelValues(skippedReasonNoopSubquery).Inc()
+		}
+		if stats.GetNoOpNonSplittableQuery() {
+			s.metrics.splittingSkipped.WithLabelValues(skippedReasonNoopNonSplittable).Inc()
+		}
```

Comment:
Note for reviewers: this means that it is possible for a query to be both:
* no op small interval
* no op subquery

For instance, ``rate(metric_counter[1m]) / rate(metric_counter[5h:5m]) > 0.5``.

However, it can only be no op non-splittable if none of the above is true. Let me know if you see any concerns with this 🤔